llvm/lib/Target/AIE/AIE2LegalizerInfo.cpp

```diff
@@ -461,6 +461,17 @@ AIE2LegalizerInfo::AIE2LegalizerInfo(const AIE2Subtarget &ST) : AIEHelper(ST) {
       .clampMaxNumElements(0, S32, 16)
       .custom();
 
+  getActionDefinitionsBuilder(G_BUILD_VECTOR_TRUNC)
+      .unsupportedIf(IsNotValidDestinationVector)
+      // We clamp the high values and not the low ones, sice the former
+      // splits the values but the latter keeps the same G_BUILD_VECTOR_TRUNC in
+      // the output instructions which causes an infinite loop since it
+      // can't reach our custom legalization code.
```

Comment:
I'm not sure I understand the comment. Do you mean we don't set a minimum element count because that would somehow trigger an infinite loop?

Reply:
Low value clamping can cause an infinite loop because each split operation still uses G_BUILD_VECTOR_TRUNC, causing the compiler to repeatedly try to legalize the same instruction.